cli/src/pcluster/aws/ec2.py

```
@@ -487,3 +487,65 @@ def run_instances(self, **kwargs):
+                "Associations": [
+                    {
+                        "Main": False,
+                        "RouteTableAssociationId": "rtbassoc-12345678901234567",
+                        "RouteTableId": "rtb-12345678901234567",
+                        "SubnetId": "subnet-12345678901234567",
+                        "AssociationState": {"State": "associated"},
+                    }
+                ],
+                "PropagatingVgws": [],
+                "RouteTableId": "rtb-12345678901234567",
+                "Routes": [
+                    {
+                        "DestinationCidrBlock": "10.0.0.0/16",
+                        "GatewayId": "local",
+                        "Origin": "CreateRouteTable",
+                        "State": "active",
+                    },
+                    {
+                        "DestinationCidrBlock": "0.0.0.0/0",
+                        "GatewayId": "igw-12345678901234567",
+                        "Origin": "CreateRoute",
+                        "State": "active",
+                    },
+                ],
+                "Tags": [
+                    {"Key": "aws:cloudformation:logical-id", "Value": "RouteTablePublic"},
+                    ...
+                ],
+                "VpcId": "vpc-12345678901234567",
+                "OwnerId": "123456789012"
+            }
+        ]
+        :param filters: The filters to apply when describing the route tables.
+        :return: The route tables that match the filters.
+        """
+        kwargs = {"Filters": filters} if filters else {}
+        return list(self._paginate_results(self._client.describe_route_tables, **kwargs))
+
+    @AWSExceptionHandler.handle_client_exception
+    def is_subnet_public(self, subnet_id):
+        """Check if a subnet is public."""
+        route_tables = self.describe_route_tables(filters=[{"Name": "association.subnet-id", "Values": [subnet_id]}])
+        if not route_tables:
+            return False
```

This should raise an Exception since lack of route tables is not an indication that the subnet is private.

---

That makes sense. Done!